### ISSUE TITLE
1.14 update

### ISSUE THREAD

**Issue by SlimeDog** [mcmonkeyprojects/Sentinel#288]
The output of command `/version` on my server is: **(PLEASE FILL IN)**
The output of command `/version citizens` on my server is: 
```
[15:02:01] [Server thread/INFO]: This server is running CraftBukkit version git-Spigot-4d24e36-5193f76 (MC: 1.14) (Implementing API version 1.14-R0.1-SNAPSHOT)
[15:02:01] [Server thread/INFO]: You are 1 version(s) behind
```
The output of command `/version citizens` on my server is: 
```
[15:02:25] [Server thread/INFO]: Citizens version 2.0.25-SNAPSHOT (build 1669)
[15:02:25] [Server thread/INFO]: Website: http://www.citizensnpcs.co
[15:02:25] [Server thread/INFO]: Authors: aPunch and fullwall
```
The output of command `/version sentinel` on my server is: 
```
[15:03:15 INFO]: Sentinel version 1.8.5 (build 251)
[15:03:15 INFO]: Author: mcmonkey
```
A pastebin.com link to my (full!) `Sentinel/config.yml` file is:
https://blackdog.straight8.com/drops/sentinel-config.yml

Without sentinel, Citizens and Denizen function as expected. Sentinel has not been updated, so it is not a complete surprise that it does not function well. This report is just to point at a few of the errors that need to be addressed. We are looking forward to the 1.14-compatible update.

On server start-up, warnings are provided for each of the four configured sentinels:
```
[14:47:42] [Server thread/WARN]: [Sentinel] Citizens installation is **very outdated** and does not contain newer useful APIs. Please update your installation of the Citizens plugin!
[14:47:42] [Server thread/WARN]: [Sentinel] Citizens installation is **very outdated** and does not contain newer useful APIs. Please update your installation of the Citizens plugin!
[14:47:42] [Server thread/WARN]: [Sentinel] Citizens installation is **very outdated** and does not contain newer useful APIs. Please update your installation of the Citizens plugin!
[14:47:42] [Server thread/WARN]: [Sentinel] Citizens installation is **very outdated** and does not contain newer useful APIs. Please update your installation of the Citizens plugin!
```
When I tried to exercise the sentinels, by spawning a target mob:
```
[14:48:18] [Server thread/INFO]: [Citizens] Exception while updating 1: null.
[14:48:18] [Server thread/WARN]: java.lang.NullPointerException
[14:48:18] [Server thread/WARN]:        at org.mcmonkey.sentinel.targeting.SentinelTargetList.recalculateTargetsCache(SentinelTargetList.java:384)
[14:48:18] [Server thread/WARN]:        at org.mcmonkey.sentinel.targeting.SentinelTargetList.checkRecalculateTargetsCache(SentinelTargetList.java:341)
[14:48:18] [Server thread/WARN]:        at org.mcmonkey.sentinel.targeting.SentinelTargetList.isTarget(SentinelTargetList.java:48)
[14:48:18] [Server thread/WARN]:        at org.mcmonkey.sentinel.targeting.SentinelTargetingHelper.isTargeted(SentinelTargetingHelper.java:222)
[14:48:18] [Server thread/WARN]:        at org.mcmonkey.sentinel.targeting.SentinelTargetingHelper.shouldTarget(SentinelTargetingHelper.java:65)
[14:48:18] [Server thread/WARN]:        at org.mcmonkey.sentinel.targeting.SentinelTargetingHelper.findBestTarget(SentinelTargetingHelper.java:419)
[14:48:18] [Server thread/WARN]:        at org.mcmonkey.sentinel.SentinelTrait.runUpdate(SentinelTrait.java:1056)
[14:48:18] [Server thread/WARN]:        at org.mcmonkey.sentinel.SentinelTrait.run(SentinelTrait.java:1282)
[14:48:18] [Server thread/WARN]:        at net.citizensnpcs.api.npc.AbstractNPC.update(AbstractNPC.java:430)
[14:48:18] [Server thread/WARN]:        at net.citizensnpcs.npc.CitizensNPC.update(CitizensNPC.java:283)
[14:48:18] [Server thread/WARN]:        at net.citizensnpcs.nms.v1_14_R1.entity.WolfController$EntityWolfNPC.mobTick(WolfController.java:201)
[14:48:18] [Server thread/WARN]:        at net.minecraft.server.v1_14_R1.EntityInsentient.doTick(EntityInsentient.java:658)
[14:48:18] [Server thread/WARN]:        at net.minecraft.server.v1_14_R1.EntityLiving.movementTick(EntityLiving.java:2450)
[14:48:18] [Server thread/WARN]:        at net.minecraft.server.v1_14_R1.EntityInsentient.movementTick(EntityInsentient.java:501)
[14:48:18] [Server thread/WARN]:        at net.minecraft.server.v1_14_R1.EntityAgeable.movementTick(EntityAgeable.java:154)
[14:48:18] [Server thread/WARN]:        at net.minecraft.server.v1_14_R1.EntityAnimal.movementTick(EntityAnimal.java:29)
[14:48:18] [Server thread/WARN]:        at net.minecraft.server.v1_14_R1.EntityWolf.movementTick(EntityWolf.java:153)
[14:48:18] [Server thread/WARN]:        at net.minecraft.server.v1_14_R1.EntityLiving.tick(EntityLiving.java:2286)
[14:48:18] [Server thread/WARN]:        at net.minecraft.server.v1_14_R1.EntityInsentient.tick(EntityInsentient.java:272)
[14:48:18] [Server thread/WARN]:        at net.minecraft.server.v1_14_R1.EntityWolf.tick(EntityWolf.java:169)
[14:48:18] [Server thread/WARN]:        at net.minecraft.server.v1_14_R1.WorldServer.entityJoinedWorld(WorldServer.java:578)
[14:48:18] [Server thread/WARN]:        at net.minecraft.server.v1_14_R1.World.a(World.java:743)
[14:48:18] [Server thread/WARN]:        at net.minecraft.server.v1_14_R1.WorldServer.doTick(WorldServer.java:355)
[14:48:18] [Server thread/WARN]:        at net.minecraft.server.v1_14_R1.MinecraftServer.b(MinecraftServer.java:1054)
[14:48:18] [Server thread/WARN]:        at net.minecraft.server.v1_14_R1.DedicatedServer.b(DedicatedServer.java:396)
[14:48:18] [Server thread/WARN]:        at net.minecraft.server.v1_14_R1.MinecraftServer.a(MinecraftServer.java:953)
[14:48:18] [Server thread/WARN]:        at net.minecraft.server.v1_14_R1.MinecraftServer.run(MinecraftServer.java:798)
[14:48:18] [Server thread/WARN]:        at java.base/java.lang.Thread.run(Thread.java:834)
[14:48:18] [Server thread/INFO]: [Citizens] Exception while updating 3: null.
```

**Comment by mcmonkey4eva** [comment]
```
[14:47:42] [Server thread/WARN]: [Sentinel] Citizens installation is **very outdated** and does not contain newer useful APIs. Please update your installation of the Citizens plugin!
```
That error should not be showing at all... that's a warning meant to show for MC 1.8 servers running a version of Citizens from the same era rather than current builds. It only even got processed as of course Sentinel doesn't recognize the version 1.14 yet. I'm not sure why the error showed though, it's driven by a check for a class that definitely exists in modern Citizens.
-- er, oh, actually there's a mixup because Java identifies classes stupidly. That's fixed now! :D 

--------

The other error is just because nothing loaded with 1.14 unrecognized.

I've made the full update to 1.14, including all basic mob types and such. There were also a few available-API changes and some trouble with arrows, but things should work fine now.
To prevent similar trouble in the future, I've added code to recognize "1.15" and "1.16" as just still being 1.14 (so things will hopefully mostly still work when the next update comes, until I get around to the actual matching updates for those).

MC 1.14 compatible builds are labeled as Sentinel version 1.9.1 or higher.

**Comment by SlimeDog** [comment]
Thanks. Note that api-version:1.14 is unrecognized by Spigot 1.13.2, so b252 will not load on Spigot 1.13.2 final.

**Comment by mcmonkey4eva** [comment]
Yikes... did not expect it would literally refuse to load for that.
I've fixed that (by which I mean: set it back to 1.13) in build 253+

**Comment by SlimeDog** [comment]
Yep. Thanks.
